internal/configuration/settings.go

```diff
@@ -124,10 +135,12 @@ type Settings struct {
 }
 
 // NewSettings creates a new Settings object with default values.
-func NewSettings(ctx context.Context, k8sClient *kubernetes.Clientset) (*Settings, error) {
+func NewSettings(ctx context.Context, k8sClient kubernetes.Interface) (*Settings, error) {
 	settings := &Settings{
-		Context:   ctx,
-		K8sClient: k8sClient,
+		Context:      ctx,
+		K8sClient:    k8sClient,
+		TlsMode:      "",
+		Certificates: nil,
```

Comment:
Are there any secrets here now that we should be sure don't show up in logs or stack traces if there is a crash or we log this struct?

Reply:
I do not believe so, created [Issue 145](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/issues/145) to address the Certificates type and make it non-loggable.